src/app/lib/utilities/getStoryPromoInfo/index.jsx

```diff
@@ -28,10 +28,15 @@ export const getHeadline = item => {
   return headline;
 };
 
-export const getUrl = item => {
+export const getUrl = (item, service = null, variant = null) => {
   const assetUri = pathOr(null, ['locators', 'assetUri'], item);
   const canonicalUrl = pathOr(null, ['locators', 'canonicalUrl'], item);
-  const uri = pathOr(null, ['uri'], item);
+  let uri = pathOr(null, ['uri'], item);
+  if (uri && service === 'zhongwen') {
```

Comment:
Let's not restrict this to zhongwen only - why don't we check for the existence of variant instead?